### PR TITLE
[ADD] OD-1192: Add pdf return labels creation logic to return pickings.

### PR DESCRIPTION
**Pull request by benrazanskas**

### CODE REVIEW

**Comment by benrazanskas** [comment]
So the thing is, i'm looking for solution to deal with vanmoof_shipping dependencies while was running OCA tests, since vanmoof_shipping has dependencies and raises lot of checks, maybe you guys have an idea how to overcome on this part. Short story: vanmoof_shipping is asking for box to be added before it lets to do smth about labels, so I had to include this here https://github.com/VanMoof/delivery-carrier/pull/5/files#diff-353f63d411d851c3fea1b07cf49d6ece34db580952e5659d16370bd41a32e687R122-R125
Maybe it's good thing to move unittests in our side modules?

File: delivery_carrier_label_paazl/models/stock_picking.py [hunk 1]
```diff
@@ -67,15 +67,26 @@ def _paazl_generate_labels(self):
...
+                payload.pop("labelFormat")
+            # Generating return labels
+            status = api.service.generatePdfReturnLabels(**payload)
+        else:
+            # we need to create labels in order to get a tracking number
+            status = api.service.generateLabels(**self._paazl_generate_labels_data())
         self._paazl_raise_error(status)
         # as we have the labels anyways, attach them
         account = self._get_carrier_account()
+        status_label = status.label if return_rma else status.labels
```
Comment (StefanRijnhart) [inline delivery_carrier_label_paazl/models/stock_picking.py:85]: Is this confirmed? It says in the PDF that I have that the generateReturnLabels request also returns `labels`
image

File: delivery_carrier_label_paazl/models/stock_picking.py [hunk 2]
```diff
@@ -67,15 +67,26 @@ def _paazl_generate_labels(self):
...
         api = self._paazl_api()
-        # we need to create labels in order to get a tracking number
-        status = api.service.generateLabels(**self._paazl_generate_labels_data())
+        is_return = self.location_id.usage == "customer"
+        return_rma = all([is_return and self.picking_type_code == "incoming"])
+        if return_rma:
+            payload = self._paazl_generate_labels_data()
+            # Removing since generatePdfReturnLabels call doesn't accept labelFormat
+            if "labelFormat" in payload:
+                payload.pop("labelFormat")
```
Comment (StefanRijnhart) [inline delivery_carrier_label_paazl/models/stock_picking.py:76]: ~~Do we need to pass the shippingOption in the payload here? Presumably this is not the same as the original order's option but an option that triggers a pickup at the customer location.~~ Ah, below I see.

File: delivery_carrier_label_paazl/models/stock_picking.py (hunk 1 above)
Comment (benrazanskas) [reply delivery_carrier_label_paazl/models/stock_picking.py:85]: Yea I know, but reponse was geting .label for me after I was making calls to generatePdfReturnLabels.

File: delivery_carrier_label_paazl/models/stock_picking.py (hunk 2 above)
Comment (benrazanskas) [reply delivery_carrier_label_paazl/models/stock_picking.py:76]: Without shippingOption pass, I wasnt able to make a call to that endpoint, error text was similar to : "Please provide at least one shipping option"

File: delivery_carrier_label_paazl/models/stock_picking.py (hunk 1 above)
Comment (StefanRijnhart) [reply delivery_carrier_label_paazl/models/stock_picking.py:85]: Alright, that settles it :grin:

**Review by StefanRijnhart — APPROVED** [review]

**Review by antonioburic — APPROVED** [review]

**Review by MPishpecki — APPROVED** [review]

**Comment by MPishpecki** [comment]
@benrazanskas we cannot deploy this with a tag. Can you please add it to the buildout the old way and inform us. Either @antonioburic or I deploy it manually or it goes with the next ticket that will be triggered for deployment.

**Comment by dkubiak789** [comment]
I added a PRs [#505](https://github.com/OCA/delivery-carrier/pull/505) to OCA/delivery-carrier and [#1192](https://github.com/VanMoof/odoo-addons/pull/1192) to buildout in odoo-addons.

**Comment by MPishpecki** [comment]
@dkubiak789 @benrazanskas is still fixing failing tests. Maybe too early for the OCA PR? If you are keeping it please make sure you update the OCA PR for the changes.

**Comment by antonioburic** [comment]
[The last commit](https://github.com/VanMoof/delivery-carrier/pull/5/commits/c5ed936d3a520cbc52f9f44db13ac8a29372db9b) addresses the feedback from testing - inverted sender/receiver address for return labels.

**Comment by pankk** [comment]
Stale upstream, but were are using the code. Due to a conflict this PR was merged into: https://github.com/VanMoof/delivery-carrier/pull/8

**Pull request closed without merging**